web/quickfeed_service.go

```diff
@@ -69,7 +67,7 @@ func (s *QuickFeedService) GetUserByCourse(_ context.Context, in *connect.Reques
 	user, err := s.db.GetUserByCourse(query, in.Msg.UserLogin)
 	if err != nil {
 		s.logger.Errorf("GetUserByCourse failed: %v", err)
-		return nil, status.Error(codes.FailedPrecondition, "failed to get student information")
+		return nil, connect.NewError(connect.CodeFailedPrecondition, errors.New("failed to get student information"))
```

Comment:
Should this be `InvalidArgument` instead?

Reply:
Or `NotFound`?

Reply:
Agree, `NotFound` makes perfect sense here.